app/models/certificate.server.ts

```diff
@@ -0,0 +1,64 @@
+import type { Certificate } from '@prisma/client';
+
+import { prisma } from '~/db.server';
+
+export type { Certificate } from '@prisma/client';
+
+export async function getCertificateByUsername(username: Certificate['username']) {
+  return prisma.certificate.findUnique({ where: { username } });
+}
+
+export async function createCertificate(
+  username: Certificate['username'],
+  subject: Certificate['subject'],
+  certificate: Certificate['certificate'],
+  orderUrl: Certificate['orderUrl'],
+  privateKey: Certificate['privateKey']
+) {
+  // Set expiration date 90 days from now
+  let validTo = new Date();
+  validTo.setDate(validTo.getDate() + 90);
```

Comment:
Is there an issue if you assigned `validTo` with `new Date().getDate() + 90)`?

Reply:
When we do that and leave `validTo` undefined, the issue is that typescript will complain about `validTo` possibly being undefined. Therefore, we would still end up with 2 lines (one to declare, and another to set date), and I still need to add code to solve the complain on top of that.

Reply:
it wouldn't be undefined no? You're assigning it a values from the start `let validTo = new Date().getDate() + 90`

Reply:
Oh, I misunderstood you. In that case, it would complain about `Type 'number' is not assignable to type 'string | Date'`.

Reply:
oh gotcha